src/syft/lib/tensor/tensorbase.py

```diff
@@ -0,0 +1,172 @@
+        return IntegerTensor(child=self.child - other.child)
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __mul__(self, other: "IntegerTensor") -> "IntegerTensor":
+        return IntegerTensor(child=self.child * other.child)
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __truediv__(self, other: Union[int, float]) -> "IntegerTensor":
+        return IntegerTensor(child=self.child / other)
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __matmul__(self, other: "IntegerTensor") -> "IntegerTensor":
+        return IntegerTensor(child=self.child @ other.child)
+
+
+class SyftTensor(ChildDelegatorTensor):
+    def __init__(self, child: Union[FloatTensor, IntegerTensor]) -> None:
+        self.child = child
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __add__(self, other: "SyftTensor") -> "SyftTensor":
+        if isinstance(self.child, FloatTensor) and isinstance(other.child, FloatTensor):
+            return SyftTensor(child=self.child + other.child)
+        elif isinstance(self.child, IntegerTensor) and isinstance(
+            other.child, IntegerTensor
+        ):
+            return SyftTensor(child=self.child + other.child)
+        else:
+            raise ValueError()
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __sub__(self, other: "SyftTensor") -> "SyftTensor":
+        if isinstance(self.child, FloatTensor) and isinstance(other.child, FloatTensor):
+            return SyftTensor(child=self.child - other.child)
+        elif isinstance(self.child, IntegerTensor) and isinstance(
+            other.child, IntegerTensor
+        ):
+            return SyftTensor(child=self.child - other.child)
+        else:
+            raise ValueError()
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __mul__(self, other: "SyftTensor") -> "SyftTensor":
+        if isinstance(self.child, FloatTensor) and isinstance(other.child, FloatTensor):
+            return SyftTensor(child=self.child * other.child)
+        elif isinstance(self.child, IntegerTensor) and isinstance(
+            other.child, IntegerTensor
+        ):
+            return SyftTensor(child=self.child * other.child)
+        else:
+            raise ValueError()
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __truediv__(self, other: Union[int, float]) -> "SyftTensor":
+        if isinstance(self.child, FloatTensor) and isinstance(other, int):
+            return SyftTensor(child=self.child / other)
+        if isinstance(self.child, IntegerTensor) and isinstance(other, float):
+            return SyftTensor(child=self.child / other)
+        else:
+            raise ValueError()
+
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def __matmul__(self, other: "SyftTensor") -> "SyftTensor":
+        if isinstance(self.child, FloatTensor) and isinstance(other, SyftTensor):
```

Comment:
Q: Why not use?
```
if isinstance(self.child, (FloatTensor, IntegerTensor))
```
In that case there will be only 1 ```if``` statement

Reply:
Oh yes forgot about that functionality

Reply:
Sorry for not giving more information!

I meant this if and the next one.
``` 
if isinstance(self.child, FloatTensor) and isinstance(other, SyftTensor):
  return SyftTensor(child=self.child @ other.child)
if isinstance(self.child, IntegerTensor) and isinstance(other, SyftTensor):
  return SyftTensor(child=self.child @ other.child)
```

Changed to:
``` 
if isinstance(self.child, (FloatTensor, IntegerTensor)) and isinstance(other, SyftTensor):
  return SyftTensor(child=self.child @ other.child)
```

Reply:
Turns out this isn't possible due to Mypy, I get the error:
```
Both left and right operands are unions
```

I tried creating a Union type but isinstance can't work with "Typing" stuff. There could be a way around this but for now I think its okay.